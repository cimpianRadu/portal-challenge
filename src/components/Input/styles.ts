import {StyleSheet} from 'react-native';
import {COLORS} from 'theme/colors';

const styles = StyleSheet.create({
  label: {
    fontSize: 14,
    color: COLORS.SECONDARY_GRAY,
    alignSelf: 'flex-start',
  },
  input: {
    width: '100%',
    height: 40,
    borderColor: COLORS.SECONDARY_GRAY,
    borderWidth: 1,
  },
  errorMessage: {
    fontSize: 12,
    color: COLORS.RED_ERRROR,
  },
});

export default styles;
