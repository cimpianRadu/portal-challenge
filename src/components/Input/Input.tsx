import React from 'react';
import {
  StyleProp,
  Text,
  TextInput,
  TextInputProps,
  TextStyle,
} from 'react-native';
import styles from './styles';

const Input = ({
  label,
  labelStyle,
  errorMessage,
  ...props
}: TextInputProps & {
  label: string;
  labelStyle?: StyleProp<TextStyle>;
  errorMessage?: string;
}) => {
  return (
    <>
      <Text style={[styles.label, labelStyle]}>{label}</Text>
      <TextInput style={styles.input} {...props} />
      {errorMessage && <Text style={styles.errorMessage}>{errorMessage}</Text>}
    </>
  );
};

export default Input;
