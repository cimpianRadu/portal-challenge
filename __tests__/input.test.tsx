import React from 'react';
import {render} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import Input from 'components/Input';

describe('Input Component', () => {
  const testLabel = 'Test Label';
  const testErrorMessage = 'Error message';

  it('renders correctly', () => {
    const {getByText} = render(<Input label={testLabel} />);
    expect(getByText(testLabel)).toBeTruthy();
  });

  it('displays the label correctly', () => {
    const {getByText} = render(<Input label={testLabel} />);
    expect(getByText(testLabel)).toBeDefined();
  });

  it('shows an error message when provided', () => {
    const {getByText} = render(
      <Input label={testLabel} errorMessage={testErrorMessage} />,
    );
    expect(getByText(testErrorMessage)).toBeDefined();
  });

  it('renders correctly and matches snapshot', () => {
    const tree = renderer.create(<Input label={testLabel} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
