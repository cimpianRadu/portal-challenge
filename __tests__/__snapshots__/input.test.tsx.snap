// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders correctly and matches snapshot 1`] = `
[
  <Text
    style={
      [
        {
          "alignSelf": "flex-start",
          "color": "gray",
          "fontSize": 14,
        },
        undefined,
      ]
    }
  >
    Test Label
  </Text>,
  <TextInput
    style={
      {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "width": "100%",
      }
    }
  />,
]
`;
